ltiauthenticator/__init__.py

```diff
@@ -131,9 +131,13 @@ def authenticate(self, handler, data=None):
         for k, values in handler.request.body_arguments.items():
             args[k] = values[0].decode() if len(values) == 1 else [v.decode() for v in values]
 
+        # handle multiple layers of proxied protocol (comma separated) and take the outermost
+        protocol = handler.request.headers['x-forwarded-proto'].split(',')[0]
```

Comment:
This will fail with an exception if the 'x-forwarded-proto' header does not exist, unfortunately!